### DISCUSSION TITLE
How to ignore reportUnnecessaryTypeIgnoreComment for a particular line?

### DISCUSSION THREAD

**Discussion by BrandonNav** [microsoft/pyright#4934] [Q&A] [answered]
Hello,

I have an import which is available in dev environments but not prod environments

e.g.

```python
try:
    import local_settings
except ImportError:
    pass
```

I have `reportMissingImports` enabled and pyright correctly identifies this as an error.

so I add `# pyright: ignore [reportMissingImports]` to the line but then I get the following error `Unnecessary "# pyright: ignore" rule: "reportMissingImports"`

I have tried `# pyright: ignore [reportMissingImports, reportUnnecessaryTypeIgnoreComment]` but that still reports an error.
I can add `#pyright: reportUnnecessaryTypeIgnoreComment=false` but would prefer to keep this setting enabled.

Is there a way to ignore the unnecessary type ignore for a particular line?

**Comment by erictraut** [comment, marked as answer]
Are you running pyright in the context of two different Python environments — both the dev and prod environments? Type checking is typically done in one or the other but not both.

If you need to run type checking in both contexts, here are some potential workarounds:
1. Use a `TYPE_CHECKING` conditional.
2. Create separate pyright configuration files for your dev and prod environments. In the prod version, disable `reportMissingImports` or `reportUnnecessaryTypeIgnoreComment` or both.
3. Move the import of this module to a separate internal source file that includes the try-wrapped import statement. By placing it in its own source file, you can add `# pyright: reportMissingImports = false` to just that file.
```python
# dev_imports.py
# pyright: reportMissingImports = false
try:
    import local_settings
except ImportError:
    pass

# otherfile.py
from dev_imports import local_settings
```

**Reply by BrandonNav** [reply]
Thanks for the response @erictraut
I'll go with your 3rd option, though 1 is also a good option.

Just for your own context: We aren't using different environments as we don't want to be surprised by CI failures (CI runs in with prod-like settings). We'd find it a little frustrating to commit with everything passing locally only for CI to fail.
